core/src/main/java/org/infinispan/statetransfer/StateTransferInterceptor.java

```diff
@@ -380,13 +386,19 @@ private BasicInvocationStage handleNonTxWriteCommand(InvocationContext ctx, Writ
          }
          if (!(ce instanceof OutdatedTopologyException) && !(ce instanceof SuspectException))
             throw t;
+         WriteCommand writeCommand = (WriteCommand) rCommand;
+         if (writeCommand.hasFlag(Flag.ZERO_LOCK_ACQUISITION_TIMEOUT)) {
+            // the command is expected to not wait
```

Comment:
State transfer lock is a lock as well :)

Anyway, I don't agree, but I can remove that change since it's not a necessary part of the PR. You can have the dangling threads/futures waiting for the topology and then failing if you like them so much.